src/client.rs

```diff
@@ -65,13 +61,16 @@ impl Plugin for ClientPlugin {
                 (ClientSet::Send, ClientSet::SendPackets).chain(),
             )
             .add_systems(Startup, setup_channels)
+            .add_systems(
+                OnExit(ClientState::Connected),
+                reset.in_set(ClientSet::Reset),
+            )
```

Comment:
This is not equivalent. `client_just_disconnected` also encompasses the connecting -> disconnected sequence.